saged/all_label_comparison.py

```diff
@@ -0,0 +1,126 @@
+"""
+This benchmark compares the performance of different models in learning to predict all the
+classes in the dataset
+"""
+import argparse
+
+import sklearn.metrics
+import yaml
+
+from saged import utils, datasets, models
+
+
+if __name__ == '__main__':
+    parser = argparse.ArgumentParser()
+    parser.add_argument('dataset_config',
+                        help='The yaml formatted dataset configuration file. For more information '
+                             'about this file read the comments in the example_dataset.yml file')
+    parser.add_argument('supervised_config',
+                        help='The yaml formatted model configuration file. For more information '
+                             'about this file read the comments in the example_model.yml file')
+    parser.add_argument('out_file',
+                        help='The file to save the results to')
+    parser.add_argument('--unsupervised_config',
+                        help='The yaml formatted model configuration file for an unsupervised '
+                             'model. If omitted, the benchmark will use the original features '
+                             'for training. For more information about this file read the '
+                             'comments in the example_model.yml file',
+                        default=None)
+    parser.add_argument('--neptune_config',
+                        help='A yaml formatted file containing init information for '
+                             'neptune logging')
+    parser.add_argument('--seed',
+                        help='The random seed to be used in the experiment',
+                        type=int,
+                        default=42)
+    args = parser.parse_args()
+
+    with open(args.dataset_config) as data_file:
+        dataset_config = yaml.safe_load(data_file)
+
+    with open(args.supervised_config) as supervised_file:
+        supervised_config = yaml.safe_load(supervised_file)
+
+    # Parse config file
+
+    if args.neptune_config is not None:
+        with open(args.neptune_config) as neptune_file:
+            neptune_config = yaml.safe_load(neptune_file)
+            utils.initialize_neptune(neptune_config)
+
+    # Get the class of dataset to use with this configuration
+    MixedDatasetClass = getattr(datasets, dataset_config['name'])
+
+    # We no longer need this variable, and it isn't used in the constructor
+    del dataset_config['name']
+
+    print('Loading all data')
+    all_data = MixedDatasetClass.from_config(**dataset_config)
+    print('Subsetting labeled data')
+    labeled_data = all_data.get_labeled()
+    print('Subsetting unlabeled data')
+    unlabeled_data = all_data.get_unlabeled()
+    print('Splitting data')
+
+    # Get fivefold cross-validation splits
+    labeled_splits = labeled_data.get_cv_splits(num_splits=5, seed=args.seed)
```

Comment:
You could make the number of splits a command-line parameter as well, or define it in a config file somewhere. I doubt you'll need to change it much, if at all, but it probably wouldn't hurt to make it more accessible.

Reply:
Will do